sourcegraph/sourcegraph
# Batch Changes: Replace MultiSelect with wildcard MultiCombobox UI

## Background

In this PR https://github.com/sourcegraph/sourcegraph/pull/46362 we added a new wildcard component for MultiCobobox-like UI. MultiSelect (react-select library) provides not so straightforward API and types around this component, it also possible that one of the react-query types slows down TypeScript compiler see related [slack thread ](https://sourcegraph.slack.com/archives/C01C3NCGD40/p1669864732998109?thread_ts=1669859170.510239&cid=C01C3NCGD40)

Also multi select isn't so small library it's 90kb minified, so In this PR we replace this component with our in-house implementation of MultiCombobox UI

<img width="1025" alt="Screenshot 2023-01-13 at 22 59 49" src="https://user-images.githubusercontent.com/18492575/212445637-363cffd4-5482-406c-9220-2f8109175cfb.png">

## Test plan
- Check Batch changes filters UI 
- Filters should be preserved in URL and also saved in temporary setting store
- Check that with new UI you still can see filters from the old UI version
- Check that list page has right set of entities based on selected filters



## App preview:

- [Web](https://sg-web-vk-replace-multi-select-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
